dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/FormUtils.java

```diff
@@ -0,0 +1,303 @@
+                os.write('&');
+            }
+
+        }
+    }
+    
+    public static void populateMapFromMultipart(MultivaluedMap<String, String> params,
+                                                MultipartBody body,
+                                                Message m,
+                                                boolean decode) {
+        List<Attachment> atts = body.getAllAttachments();
+        checkNumberOfParts(m, atts.size());
+        for (Attachment a : atts) {
+            ContentDisposition cd = a.getContentDisposition();
+            if (cd != null && !MULTIPART_FORM_DATA_TYPE.equalsIgnoreCase(cd.getType())) {
+                continue;
+            }
+            String cdName = cd == null ? null : cd.getParameter("name");
+            String contentId = a.getContentId();
+            String name = StringUtils.isEmpty(cdName) ? contentId : cdName.replace("\"", "").replace("'", "");
+            if (StringUtils.isEmpty(name)) { 
+                throw ExceptionUtils.toBadRequestException(null, null);
+            }
+            if (CONTENT_DISPOSITION_FILES_PARAM.equals(name)) {
+                // this is a reserved name in Content-Disposition for parts containing files
+                continue;
+            }
+            try {
+                String value = IOUtils.toString(a.getDataHandler().getInputStream());
+                params.add(HttpUtils.urlDecode(name),
+                           decode ? HttpUtils.urlDecode(value) : value);
+            } catch (IllegalArgumentException ex) {
+                LOG.warning("Illegal URL-encoded characters, make sure that no "
+                    + "@FormParam and @Multipart annotations are mixed up");
+                throw ExceptionUtils.toInternalServerErrorException(ex, null);
+            } catch (IOException ex) {
+                throw ExceptionUtils.toBadRequestException(null, null);
+            }
+        }
+    }
+    
+    private static void checkNumberOfParts(Message m, int numberOfParts) {
+        if (m == null || m.getExchange() == null || m.getExchange().getInMessage() == null) {
+            return;
+        }
+        String maxPartsCountProp = (String)m.getExchange()
+            .getInMessage().getContextualProperty(MAX_FORM_PARAM_COUNT);
+        if (maxPartsCountProp == null) {
+            return;
+        }
+        try {
+            int maxPartsCount = Integer.parseInt(maxPartsCountProp);
+            if (maxPartsCount != -1 && numberOfParts >= maxPartsCount) {
+                throw new WebApplicationException(413);
+            }
+        } catch (NumberFormatException ex) {
+            throw ExceptionUtils.toInternalServerErrorException(ex, null);
+        }
+    }
+
+    public static boolean isFormPostRequest(Message m) {
+        // Liberty Change Start
+        String contentType = (String) m.get(Message.CONTENT_TYPE);
+        
+        if (contentType != null) {
+            int indx = contentType.indexOf(';');
```

Comment:
instead of carving off the charset, could you do something like:

```
return (contentType != null && contentType.startsWith(MediaType.APPLICATION_FORM_URLENCODED) &&
        HttpMethod.POST.equals(m.get(Message.HTTP_REQUEST_METHOD);
```

also, could the content type in the message use different casing than value from MediaType?  I think MediaType is always lowercase, but I'm not sure about what is used in the Message...